api/src/stampy_chat/chat.py

```diff
@@ -141,7 +141,30 @@ def construct_prompt(query: str, mode: str, history: List[Dict[str, str]], conte
 import time
 import json
 
-def talk_to_robot_internal(index, query: str, mode: str, history: List[Dict[str, str]], k: int = STANDARD_K, log: Callable = print):
+
+def check_openai_moderation(prompt: Prompt, query: str, log: Callable):
+    # 3. Run both the standalone query and the full prompt through
+    # moderation to see if it will be accepted by OpenAI's api
+
+    prompt_string = '\n\n'.join([message["content"] for message in prompt])
+    mod_res = openai.Moderation.create( input = [ query, prompt_string ])
+
+    if any(map(lambda x: x["flagged"], mod_res["results"])):
+        logger.moderation_issue(query, prompt_string, mod_res)
+
+        raise ValueError("This conversation was rejected by OpenAI's moderation filter. Sorry.")
+
+
+def remaining_tokens(prompt: Prompt):
+    # Count number of tokens left for completion (-50 for a buffer)
+    used_tokens = sum([
+        len(ENCODER.encode(message["content"]) + ENCODER.encode(message["role"]))
+        for message in prompt
+    ])
+    return NUM_TOKENS - used_tokens - TOKENS_BUFFER
+
+
+def talk_to_robot_internal(index, query: str, mode: str, history: List[Dict[str, str]], k: int = STANDARD_K, log: Callable = logger.info):
```

Comment:
```python
log: Callable = logger.info
```
There's a few places across the codebase following this pattern. I understand why I wrote it that way when the fallback logger was `print`, but with your new logging changes does it still make sense to pass in a log function with a default parameter? If so, why?

Reply:
no. I left it so there were as few changes as possible - the next PR removes them

Reply:
Sounds good, though in the future it would be better to do changes like that on your first PR then pull them into the second.